instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/tracer_middleware.rb

```diff
@@ -93,7 +93,9 @@ def call(env) # rubocop:disable Metrics/MethodLength
           private
 
           def untraced_request?(env)
-            return true if @untraced_endpoints.include?(env['PATH_INFO'])
+            return true if @untraced_endpoints.any? do |endpoint|
+              endpoint.is_a?(Regexp) ? endpoint.match?(env['PATH_INFO']) : endpoint == env['PATH_INFO']
+            end
             return true if config[:untraced_requests]&.call(env)
```

Comment:
Curious why using the existing `untraced_requests` config option that allows for lambdas is insufficient here?

Added a test block to the existing rack instrumentation, and it appears it allows for the functionality you're looking to add.
```rb
        describe 'example using lambda' do
          let(:untraced_callable) do
            ->(env) { %r{\A/ping/?}.match?(env['PATH_INFO']) }
          end

          it 'does not trace things that match the regex' do
            Rack::MockRequest.new(rack_builder).get('/ping', env)

            ping_span = finished_spans.find { |s| s.attributes['http.target'] == '/ping' }
            _(ping_span).must_be_nil

            root_span = finished_spans.find { |s| s.attributes['http.target'] == '/' }
            _(root_span).wont_be_nil
          end
        end
      end
```

Reply:
Yep, I am doing same the currently, but its a lot simpler/less boilerplate in config to add a Regexp.

Reply:
> why not remove `untraced_endpoints` entirely and have users do this?

It's a great idea. We didn't do that because we implemented `untraced_endpoints` first and we didn't want to break anyone who was already using it. It's also possible to configure `untraced_endpoints` via environment variable, but `callable` options can't be configured that way: https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/174e0fb80955522700f86671f1d156e797075e01/instrumentation/base/lib/opentelemetry/instrumentation/base.rb#L358-L362

Reply:
> It's a great idea

Hmmm... IMHO `untraced_endpoints` covers 80%+ of use cases here in a simpler fashion, I think its a good reason to keep it as its own option.